src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift

```diff
@@ -266,10 +271,7 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
 				guard let self = self, let coronaTest = self.model.coronaTest else { return }
 
 				guard coronaTest.isSubmissionConsentGiven else {
-					self.model.coronaTestService.createCoronaTestEntryInContactDiary(
-						coronaTestType: self.model.coronaTestType
-					)
-					self.showTestResultScreen()
+					self.showTestResultScreen(saveToContactDiary: true)
```

Comment:
I think it was better the other way around.
Because just by reading "self.showTestResultScreen(saveToContactDiary: true)"
Its not clear to me what's happening.
Is the test result screen saved to the contact diary?

Reply:
@ArturFriesen  you mean moving the logic from this function to where ever it was called?

Reply:
I mean that, self.showTestResultScreen should only show the screen (single responsibility). And the call for creating the test entry should not be contained there. But called separately. Like it was before (the deleted lines 269-272).

Reply:
@ArturFriesen  done and pushed